src/network/netdev/macsec.c

```diff
@@ -216,11 +210,9 @@ static int macsec_transmit_association_new_static(MACsec *s, const char *filenam
 
         security_association_init(&a->sa);
 
-        r = ordered_hashmap_ensure_allocated(&s->transmit_associations_by_section, &network_config_hash_ops);
-        if (r < 0)
-                return r;
-
-        r = ordered_hashmap_put(s->transmit_associations_by_section, a->section, a);
+        r = ordered_hashmap_ensure_put(&s->transmit_associations_by_section, &network_config_hash_ops, a->section, a);
+        if (r == -ENOMEM)
+                return log_oom();
```

Comment:
this function so far was one that doesn't log about errors, but let's the caller do this, afaics.  hence, it shouldn't log on ENOMEM now here too. 

i.e. our functions should either be of the "logging kind" or of the "non-logging kind". i.e. log about all errors they encounter, or about none (above LOG_DEBUG log level that is) — in the latter case the caller can do that instead.

So far, macsec_transmit_association_new_static() afaics is of the non-logging kind, and hence shouldn't log here either.

macsec_receive_channel_verify() otoh seems to be of the logging kind, hence everything is in order there, with the new log_oom() being inserted.

(yes, we don't get this right in many cases, and that should be fixed sooner or later. Let's fix this whenever we touch code though, as we do here...)

Reply:
OK thanks reverted.

Reply:
Not addressed yet...